Last minute Microshift RN edits/additions (#66889)
openshift/openshift-docs: microshift_release_notes/microshift-4-14-release-notes.adoc

* Previously, KCM and KAS did not exit after getting a cancellation signal. With this update, {microshift-short} shuts down as expected. (link:https://issues.redhat.com/browse/OCPBUGS-18773[*OCPBUGS-18773*])

* Previously, kubeconfigs used a certificate authority (CA) bundle that included 3 different signers: service network, localhost, and external. This made kubeconfigs interchangeable and capable of validating for other networks. With this update, the CAs used to generate kubeconfig files for {microshift-short}'s embedded components were reconfigured to ensure the kubeconfigs are independent. (link:https://issues.redhat.com/browse/OCPBUGS-8301[*OCPBUGS-8301*])

[discrete]
[id="microshift-4-14-networking-bug-fixes"]
==== Networking

* Previously, mixing uppercase and lowercase letters in the hostname caused {microshift-short} to fail. This was because the node name was taken directly from the hostname, and Kubernetes does not use uppercase names. With this update, the hostname only uses lowercase host names. (link:https://issues.redhat.com/browse/OCPBUGS-8411[*OCPBUGS-8411*])

> * Previously, the networking smoke test created a `hello-microshift` pod using `busybox` and an image pulled from Docker. With this update, quay.io is used to host the image. (link:https://issues.redhat.com/browse/OCPBUGS-19939[*OCPBUGS-19939*])

Reviewer: This is a fix related to internal e2e tests. I'm not sure we'd want it in the list.

* Previously, after changing the IP adddress, previous `kubeconfig` files for the previous IP address remained. These `kubeconfig` files were unusable because they targeted an IP address not used by {microshift-short}. With this update, the `kubeconfig` files associated with the previous IP address are deleted when the IP address changes. (link:https://issues.redhat.com/browse/OCPBUGS-15740[*OCPBUGS-15740*])

* Previously, the `microshift show-config` command did not show configuration changes. With this update, the behavior of the `microshift show-config` command shows the current configuration by default. (link:https://issues.redhat.com/browse/OCPBUGS-13221[*OCPBUGS-13221*])

* Previously, KCM and KAS did not exit after getting a cancellation signal. With this update, {microshift-short} shuts down as expected. (link:https://issues.redhat.com/browse/OCPBUGS-18773[*OCPBUGS-18773*])

> * Previously, kubeconfigs used a certificate authority (CA) bundle that included 3 different signers: service network, localhost, and external. This made kubeconfigs interchangeable and capable of validating for other networks. With this update, the certificate authorities used to generate kubeconfig files for {microshift-short}'s embedded components were reconfigured to ensure the kubeconfigs are independent. (link:https://issues.redhat.com/browse/OCPBUGS-8301[*OCPBUGS-8301*])

Reviewer 2: ```suggestion
* Previously, kubeconfigs used a certificate authority (CA) bundle that included 3 different signers: service network, localhost, and external. This made kubeconfigs interchangeable and capable of validating for other networks. With this update, the CAs used to generate kubeconfig files for {microshift-short}'s embedded components were reconfigured to ensure the kubeconfigs are independent. (link:https://issues.redhat.com/browse/OCPBUGS-8301[*OCPBUGS-8301*])
```

[discrete]
[id="microshift-4-14-networking-bug-fixes"]
==== Networking

> * Previously, mixing upper and lower case letters in the hostname makes {microshift-short} fail. This is because the node name was taken directly from the hostname, and Kubernetes does not allow upper case names. With this update, the hostname only uses lower case host names. (link:https://issues.redhat.com/browse/OCPBUGS-8411[*OCPBUGS-8411*])

Reviewer 2: ```suggestion
* Previously, mixing uppercase and lowercase letters in the hostname caused {microshift-short} to fail. This was because the node name was taken directly from the hostname, and Kubernetes does not use uppercase names. With this update, the hostname only uses lowercase host names. (link:https://issues.redhat.com/browse/OCPBUGS-8411[*OCPBUGS-8411*])
```

> * Previously, the networking smoke test created a `hello-microshift` pod using `busybox` and an image pulled from docker. With this update, quay.io is used to host the image. (link:https://issues.redhat.com/browse/OCPBUGS-19939[*OCPBUGS-19939*])

Reviewer 2: ```suggestion
* Previously, the networking smoke test created a `hello-microshift` pod using `busybox` and an image pulled from Docker. With this update, quay.io is used to host the image. (link:https://issues.redhat.com/browse/OCPBUGS-19939[*OCPBUGS-19939*])
```

> * Previously, when IP addresses were used instead of hostnames to connect to the apiserver, external connectivity failed. As a result, certificates did not regenerate when the IP address was changed. The kubeconfig used names instead of IP addresses which required the use of `subjectAltNames` if IP addresses were alternativly used. With this update, the previous kubeconfigs for absent IP addresses and names are removed. (link:https://issues.redhat.com/browse/OCPBUGS-11395[*OCPBUGS-11395*])

Reviewer 2: ```suggestion
* Previously, when IP addresses were used instead of hostnames to connect to the APIServer, external connectivity failed. The `kubeconfig` files used names instead of IP addresses which required the use of `subjectAltNames` if IP addresses were used instead of hostnames. As a result, CAs did not regenerate when an IP address was changed.  With this update, the previous `kubeconfigs` files for absent IP addresses and hostnames are removed. (link:https://issues.redhat.com/browse/OCPBUGS-11395[*OCPBUGS-11395*])
```

> * Previously, an internal apiserver IP address was configured in a loopback device to fix certificate issues. As a result, OVN-Kubernetes picks up this virtual IP address as an apiserver backend making it unreachable. With this update, use a different virtual IP address to configure a loopback device. (link:https://issues.redhat.com/browse/OCPBUGS-8277[*OCPBUGS-8277*])

Reviewer 2: ```suggestion
* Previously, an internal APIServer IP address was configured in a loopback device to fix certificate issues. As a result, OVN-Kubernetes picked up this virtual IP address as an APIServer backend making it unreachable. With this update, [something missing here?] use[s] a different virtual IP address to configure a loopback device. (link:https://issues.redhat.com/browse/OCPBUGS-8277[*OCPBUGS-8277*])
```
